### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by RossKen**

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+name: 'Feature request'
+about: Suggest an idea for improving Splink
+title: '[FEAT] <title>'
+labels: 'enhancement'
+assignees: ''
+---
```
Comment (ADBond) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: These fields probably need updating

Comment (RossKen) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Good spot. This is why templates are good - to stop idiots like me messing things up!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+<!--
+  Provide a clear and concise description of what you want to happen.
+-->
+
+### PR Checklist
+
+- [ ] Add documentation for changes
+- [ ] Add feature to example notebooks at tutorials in [splink_demos](https://github.com/moj-analytical-services/splink_demos) (if appropriate)
+- [ ] Add tests (if appropriate)
+- [ ] Changes are based off the latest version of Splink
```
Comment (ThomasHepworth) [inline .github/PULL_REQUEST_TEMPLATE.md:25]: I think these should be in the past tense.

For example, `Added documentation for changes`.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
```
Comment (ThomasHepworth) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: Do we want a "have you linted?" tickbox? Primarily to remind people to lint before they open a PR.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+---
+name: 'Feature request'
+about: Making a change to Splink
+title: '<title>'
```
Comment (ThomasHepworth) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: Do we want to start using some naming conventions that are dependent on the type of changes being proposed?

`[BUG]`, `[FEAT]`, `[MAINT]`, etc?

These could also be a ticklist item.

Comment (RossKen) [reply .github/PULL_REQUEST_TEMPLATE.md:4]: Have added checklist - feel free to add any more sections that you feel may be relevant

**Pull request merged**